Adds a New Map: Desert Outpost AKA "Dessert"

BurgerLUA opened the pull request:
## About The Pull Request

It's a map. It's small. Takes place in the desert. It's a building and a landing zone surrounded by xeno hive. Enjoy.
![image](https://user-images.githubusercontent.com/8602857/68827074-af39a980-0655-11ea-8942-cadfe46a1520.png)

![image](https://user-images.githubusercontent.com/8602857/68827008-5702a780-0655-11ea-86dc-6daaf546749a.png)
![image](https://user-images.githubusercontent.com/8602857/68827028-74d00c80-0655-11ea-8016-d5c2a66cc217.png)
![image](https://user-images.githubusercontent.com/8602857/68827042-8c0efa00-0655-11ea-97a9-66dcaa25f2fe.png)



## Why It's Good For The Game

New map good.

## Changelog
:cl: BurgerBB
add: Adds a new map: Desert Outpost
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

BurgerLUA commented:
say -1 if you're gay

tgstation-server commented:
You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

BurgerLUA commented:
fuck lmao it counts unmerged prs, suck my fat cock

Rohesie commented:
The concept is good.
What fog are you using here?

spookydonut commented on tgmc.dme:
```diff
@@ -11,6 +11,7 @@
 // END_PREFERENCES
 // BEGIN_INCLUDE
 #include "_maps\_basemap.dm"
+#include "_maps\map_files\dessert\dessert.dmm"
```
no no no no

spookydonut replied:
add it to basemaps.dm instead

XElectricX commented:
I give it my unga seal of approval.

Rohesie commented:
The map has IMDPs in it. They need to be removed, they are a balance concern.

Pariah919 commented:
Remove the IMDPs and find a way to make this map crash only possibly. Also make the outer edges have a bit less empty space in them? Perhaps make them into a cave system?

BurgerLUA commented:
Removed the PMCs because apparently xenos are unrobust and can't handle a survivor with a gun. I fixed the shutters because I thought the shutters would've went down automatically within a reasonable time but apparently that's not not a thing.

Rohesie commented:
Issues encountered on testing:
1) Outer layer of shutters are LZ 1 shutters with no visible button to make them go up. They only go up very late into the round, when both sides have been waiting for a long time.
2) Inner layer of shutters in LZ 1 are the LZ 2 shutters. Besides the name mismatch they are fine, but unlike other places where they give the marines room to build a few defenses, these don't really allow for anything. You might want to consider making the LZ 1 in the middle of the construction structures, and covering them up with the LZ 1 shutters and button.
3) Marines can't build in the sand. If this is intended, then LZ 1 should be a little closer to FOB, like point 2 suggests.
4) The room between LZ 1 and the buildings is a huge straight hallway, prime for Crusher memes, and really bad for any other xenos.
5) The whole map has vast expanses of open fields which bring multiple problems. For one, the lack of props and whatnot make the map feel very empty. It's also a sniper paradise, pretty bad for Crushers and Bulls, and in general not very interesting for xenos. They need something like Icy Caves have on their own field.
6) Some Crash fogs are too small, allowing for fire nades to burn down silos.

BurgerLUA commented:
most of this is already solved in my previous commit

Rohesie commented:
Which of these points are addressed and which are not?

BurgerLUA commented:
1, 2, 3, and 6 are fixed, with 5 being partially fixed.  4 and 5 are just opinion and will remain unchanged.

Rohesie commented:
The backup still needs to be removed.

TiviPlus commented:
Time to write an essay about the stuff that currently sucks

1) The disks are VERY easy to defend
-All three disks have 1 wide entrances with no windows or meltable walls to go through to flank, and the sci disk is even behind 2 such 1-wide doors. While you could use a crusher to take out the walls at the green disk, the other red one doesn't have enough space near it for a crusher charge. At the blue disk you would have to charge through two layers of walls to get into the disk.
- The main facility disks mentioned before are also very close together so you can easily do them at the same time by just cading across the hallway
-The fact that the Canterbury can crash 9 Tiles away from the first disk is pretty self explanatory as to why it is terrible

2) The Outer Ring
-The 1 tile space is terrible to fight in for everyone
- Mostly unused and ignored due to being a deathtrap

3) Secure storage
-Very, VERY powerful if it's not melted by xenos, meaning that less time is spent building defences and more wasted melting it
-Survivors can just waltz in and set up their own personal autism fort with unlimited plasteel, two sentries and an MG

4) Few APCs
-Repair one and half the map gets light, making enabling power (Which is also very easy to get to)

5) Not being able to flank
- As said before with the disks, the map is very linear due to the copius amounts of invincible walls and straight paths, meaning there is pretty much no way to flank for either side and making fighting predictable/stale.

BurgerLUA commented:
1. Fixed.
2. Don't even know what you're talking about here.
3. Fixed.
4. False.
5. Fixed.

Firefox13 commented:
![изображение](https://user-images.githubusercontent.com/57810301/68965721-8f3cee00-07ed-11ea-8ca8-e3e028249fb9.png)
Self-explanationary
![изображение](https://user-images.githubusercontent.com/57810301/68965838-ca3f2180-07ed-11ea-9048-2c51eacbbf38.png)
Please, no holes.
![изображение](https://user-images.githubusercontent.com/57810301/68965901-f490df00-07ed-11ea-927e-d64161534183.png)
Self-explanationary
![изображение](https://user-images.githubusercontent.com/57810301/68965970-17bb8e80-07ee-11ea-9f82-50b965bb4393.png)
shelves will look much better
...
Too many PMC corpses by the way

BurgerLUA commented:
can you not take shit quality super cropped screenshots because it's hard to find half of these bugs

BurgerLUA commented:
2 of these are irrelevant, the rest are fixed

Fikou commented:
the area names are all lowercase and it looks really dumb

Mr-Worth commented:
![need more podlock](https://user-images.githubusercontent.com/47805468/68988685-10958000-07f0-11ea-8899-27eb9675a897.PNG)
Needs more podlock

BurgerLUA commented:
That's an outdated version of the map. The current version fixed that bug.

Rohesie commented on code/game/objects/machinery/sentries.dm:
```diff
@@ -1088,6 +1086,15 @@
+	turret_flags = TURRET_LOCKED|TURRET_ON|TURRET_BURSTFIRE|TURRET_IMMOBILE
+
+/obj/machinery/marine_turret/premade/dumb/hostile/attack_hand(mob/living/user)
+	to_chat(user,"<span class='warning'>\The [src.name] refuses to cooperate!</span>")
```
```suggestion
	to_chat(user,"<span class='warning'>\The [src] refuses to cooperate!</span>")
```

BurgerLUA replied:
That is not how it works. Instead it will be The the sentry refuses to coperate. On all coderbases, it uses \The [src.name].

Rohesie commented:
The map has improved a lot. On Crash I'd say it fills the role pretty well.
Let me know when you are comfortable with it on Distress.

tgstation-server commented:
Your Fix/Feature pull request delta is currently below zero (-4). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.
